### ISSUE TITLE
why the driver pending

### ISSUE THREAD

**Issue by zeje** [asyncer-io/r2dbc-mysql#118]
**Describe the bug**
i just use mirromutth/r2dbc-mysql 0.9.x，and i read this repository code history，find none improve about connection dispose

**To Reproduce**
build 300 tables and insert 1w rows into per table.
now, provide a api, to migration them to another database.
when i run the api，it will report exception，because per request keep a connection and no release。

it will reproduce : why the driver pending and all the request pending

**Expected behavior**
i can use mysql drive to migration the data success

**Screenshots**

**Additional context**
```
@Parameters({
        @Parameter(name = "mysqlHost", in = ParameterIn.QUERY, example = "127.0.0.1"),
        @Parameter(name = "mysqlPort", in = ParameterIn.QUERY, example = "3306"),
        @Parameter(name = "username", in = ParameterIn.QUERY, example = "root"),
        @Parameter(name = "password", in = ParameterIn.QUERY, example = "123456"),
        @Parameter(name = "migrationType", in = ParameterIn.QUERY, example = "all"),
        @Parameter(name = "startTime", in = ParameterIn.QUERY, example = "1682212299000")
    })
    @GetMapping(value = "/katchu", produces = MediaType.TEXT_EVENT_STREAM_VALUE)
    @Operation(summary = " ", description = " ")
    public Flux<MigrationResult> migrationData(
        @Parameter(hidden = true) MigrationQuery query) {
              doMigrationActive(query)
    }
```

```
    private Flux<MigrationResult> doMigrationActive(MigrationQuery query) {
        ActiveService activeService = new ActiveService(query);
        return activeService.migration(query);
    }
```

```
private final static String flag = "----";
public Flux<MigrationResult> migration(MigrationQuery query) {
        return getTableNames(fixStartTime(query.getStartTime()))
            .collectList()
            .flatMapMany(it -> {
                String msg = String.format("%s表需同步的表的数量:%d",
                    getTableName(),
                    it.size());
                log.warn(msg);
                it.add(0, flag + msg);
                return Flux.fromIterable(it);
            })
            .delayElements(Duration.ofSeconds(1))
            .flatMapSequential(tableName -> {
                if (tableName.startsWith(flag)) {
                    return Flux.just(MigrationResult.finish(tableName));
                } else {
                    return migration(tableName, query.getStartTime());
                }
            }, 1);
    }

    public Flux<MigrationResult> migration(String tableName, Long startTime) {
        log.info(tableName);
        return list(tableName, startTime)
            .doOnSubscribe(sub -> {
                log.info( tableName + " migration start ");
            })
            .doFinally(signalType -> {
                log.info(tableName + " got datas ");
            })
            .doOnError((err) -> {
                err.printStackTrace();
            })
            .buffer(100)
            .publishOn(Schedulers.single())
            .concatMap(mapList -> {
                return Flux
                    .fromIterable(mapList)
                    .flatMap(objectMap -> {
                        return convert(objectMap)
                            .flatMap(message -> {
                                // insert to other database
                            });
                    })
                    .reduce(Math::addExact)
                    .map(it -> MigrationResult.success(it, tableName));
            })
            .onErrorResume(err -> LocaleUtils.resolveThrowable(err, (throwable, s) -> MigrationResult.error(s)))
            .switchIfEmpty(Mono.defer(() -> {
                String msg =  tableName + " no match data ";
                log.info(msg);
                return Mono.just(MigrationResult.error(msg));
            }))
            .doFinally(signalType -> {
                log.info(tableName + " finish migration ");
            });
    }
```

**Comment by jchrys** [comment]
Could you share your r2dbc configuration?

**Comment by zeje** [comment]
i use hsweb-easyorm, it depend on r2dbc , most of them are default

    pool:
      max-size: 32
      max-idle-time: 2m 
      max-life-time: 10m
      acquire-retry: 3

**Comment by zeje** [comment]
org.springframework.dao.DataAccessResourceFailureException: Failed to obtain R2DBC Connection; nested exception is io.r2dbc.spi.R2dbcTimeoutException: Connection acquisition timed out after 10000ms
	at org.springframework.r2dbc.connection.ConnectionFactoryUtils.lambda$getConnection$0(ConnectionFactoryUtils.java:88)
	at reactor.core.publisher.Mono.lambda$onErrorMap$31(Mono.java:3811)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106)
	at reactor.core.publisher.FluxRetry$RetrySubscriber.onError(FluxRetry.java:95)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106)
	at reactor.core.publisher.Operators.error(Operators.java:198)
	at reactor.core.publisher.MonoError.subscribe(MonoError.java:53)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4490)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103)
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onError(FluxContextWrite.java:121)
	at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124)
	at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124)
	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.handleTimeout(FluxTimeout.java:295)
	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.doTimeout(FluxTimeout.java:280)
	at reactor.core.publisher.FluxTimeout$TimeoutTimeoutSubscriber.onNext(FluxTimeout.java:419)
	at reactor.core.publisher.FluxOnErrorReturn$ReturnSubscriber.onNext(FluxOnErrorReturn.java:162)
	at reactor.core.publisher.MonoDelay$MonoDelayRunnable.propagateDelay(MonoDelay.java:271)
	at reactor.core.publisher.MonoDelay$MonoDelayRunnable.run(MonoDelay.java:286)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.r2dbc.spi.R2dbcTimeoutException: Connection acquisition timed out after 10000ms
	at io.r2dbc.pool.ConnectionPool.lambda$null$11(ConnectionPool.java:168)
	at reactor.core.publisher.Mono.lambda$onErrorMap$30(Mono.java:3797)
	at reactor.core.publisher.Mono.lambda$onErrorResume$32(Mono.java:3887)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94)
	... 17 more
Caused by: java.util.concurrent.TimeoutException: Did not observe any item or terminal signal within 10000ms in 'Connection acquisition from [io.asyncer.r2dbc.mysql.MySqlConnectionFactory@28f80a]' (and no fallback has been configured)
	... 14 more

**Comment by jchrys** [comment]
I don't think given information is enough to me to investigate further but I can give you general idea on how to handle r2dbc-mysql's conection.

In the r2dbc-mysql current implementation, MySqlConnection represents a physical connection. Once a MySqlConnection is disconnected or disposed, it cannot be reused, and you need to drop it and create a new connection to continue using the database. Normally, when using r2dbc-pool, this process is handled automatically by the connection pool.

To resolve the issue you're experiencing, make sure to properly dispose or disconnect the connection once you are done using it. This will allow the connection to be returned to the pool so that other components or requests can borrow it.

By ensuring that you handle the connection lifecycle correctly, you should be able to avoid issues with pending requests and improve the overall performance and stability of your migration process.

**Comment by zeje** [comment]
I don't know is it a bug of  r2dbc-pool or r2dbc-mysql.

the same migrate job，when I use r2dbc-postgres, it work fine，but r2dbc-mysql（also asyncer-io）

mysql data -> postgrel data -> r2dbc-pool -> r2dbc-postgres -> same code -> ✔
mysql data -> r2dbc-pool -> r2dbc-mysql -> same code -> ×

now, when i need to migrate data of mysql，i use another solution（C# entity framework core） to do it，and done。

i can just provide a test case to reproduce the problem.

if you don’t think it is a problem of r2dbc-mysql, just close this issuse.

**Comment by jchrys** [comment]
Could you provide us minimal sized test case to reproduce the problem? it would be great helpful to us to get this project forward! 😀

**Comment by jchrys** [comment]
Hello, @zeje.
Could you check if PR #119 (or latest version) resolves your issue?

**Comment by jchrys** [comment]
closing this issue, since there's no more feedback

**Comment by ssharan27** [comment]
Hi, @jchrys, I am facing this issue on Production as well, the application fails to acquire new connection after connection closed. From logs, I can see the "Connection closed by peer" message. The queries that execute after this connection closure are failing while attempting to acquire new connection causing query execution to fail.  I tried multiple configuration tweaks like changing connection validation depth to "remote", increased acquire-retries to 3 but it did not work.

As @zeje has also mentioned above

> To Reproduce
build 300 tables and insert 1w rows into per table.
>
If I understand this correctly, issue seems to be visible at scale only. I am also not seeing it on local where the load on DB is low but when I connect to DB on Production where the load is very high, this error starts showing up.

I have set tcpKeepAlive to true and maxLifeTime is not configured (defaults to no timeout). I expect the application to retain and reuse the connections it makes to DB. Even in the cases where the connection does get closed, instead of throwing exception and erroring out, shouldn't the r2dbc library attempt to reconnect and then execute the query using this new connection?

Here is the various configuration that I have tried to no avail (property names should be self expalanatory): 
```            
     ## initial config
        connectTimeout: 500ms
        maxPoolSize: 5

     ## Config 2, did not help
        connectTimeout: 500ms
        tcpKeepAlive: true
        minIdle: 5
        validationDepth: remote
        maxValidationTime: 100ms
        maxPoolSize: 5
        maxIdleTimeMS: -1
        acquireRetries: 3
```
Here is the stack trace of the error:

```
2023-06-14 00:00:39.720 WARN  [64889e5cab47bda1f51c97136795cf23,f51c97136795cf23] [reactor-tcp-epoll-2] i.a.r.mysql.client.MessageDuplexCodec    : Connection has been closed by peer |xx|tid::=-|||rid::=-|xx|
2023-06-14 00:00:39.720 WARN  [64889e5cab47bda1f51c97136795cf23,f51c97136795cf23] [reactor-tcp-epoll-2] i.a.r.mysql.client.MessageDuplexCodec    : Connection has been closed by peer |xx|tid::=-|||rid::=-|xx|
2023-06-14 00:00:39.720 WARN  [64889e5cab47bda1f51c97136795cf23,f51c97136795cf23] [reactor-tcp-epoll-2] i.a.r.mysql.client.MessageDuplexCodec    : Connection has been closed by peer |xx|tid::=-|||rid::=-|xx|
2023-06-14 00:00:39.769 WARN  [64889e5cab47bda1f51c97136795cf23,f51c97136795cf23] [reactor-tcp-epoll-2] i.a.r.mysql.client.MessageDuplexCodec    : Connection has been closed by peer |xx|tid::=-|||rid::=-|xx|
2023-06-14 00:18:09.687 INFO  [6488b9e9cea8a800073696d01b51d24e,073696d01b51d24e] [http-nio-8080-exec-7] c.i.p.g.http.HttpTriggerConfiguration    : Starting request processing |xx|tid::=-|||rid::=-|xx|
2023-06-14 00:18:09.689 INFO  [6488b9e9cea8a800073696d01b51d24e,073696d01b51d24e] [http-nio-8080-exec-7] c.i.p.g.execution.TaskExecutorService    : Processing requests: [fetch] |xx|tid::=-|||rid::=-|xx|
2023-06-14 00:18:09.694 DEBUG [6488b9e9cea8a800073696d01b51d24e,073696d01b51d24e] [customBoundedElastic-214] c.i.p.r.m.q.s.MySqlService     : Running mysql query Query{criteria=Optional[], columns=[], sort=UNSORTED, limit=-1, offset=-1} |xx|tid::=-|||rid::=-|xx|
2023-06-14 00:18:09.698 ERROR [64889e5cab47bda1f51c97136795cf23,f51c97136795cf23] [reactor-tcp-epoll-2] c.i.p.g.s.RequestExecutorService : Encountered error for request fetch |xx|tid::=-|||rid::=-|xx|
org.springframework.dao.DataAccessResourceFailureException: Failed to obtain R2DBC Connection
        at org.springframework.r2dbc.connection.ConnectionFactoryUtils.lambda$getConnection$0(ConnectionFactoryU
tils.java:90)
        at reactor.core.publisher.Mono.lambda$onErrorMap$28(Mono.java:3773)
        at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94)
        at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106)
        at reactor.core.publisher.FluxRetry$RetrySubscriber.onError(FluxRetry.java:95)
        at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:241)
        at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:315)
        at io.r2dbc.pool.MonoDiscardOnCancel$MonoDiscardOnCancelSubscriber.onError(MonoDiscardOnCancel.java:98)
        at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onError(MonoIgnoreThen.java:278)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:231)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:203)
        at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete(FluxDoFinally.java:128)
        at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete(FluxDoFinally.java:128)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:209)
        at reactor.core.publisher.Operators.complete(Operators.java:137)
        at reactor.netty.FutureMono.doSubscribe(FutureMono.java:122)
        at reactor.netty.FutureMono$DeferredFutureMono.subscribe(FutureMono.java:114)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:240)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:203)
        at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onComplete(Operators.java:2205)
        at reactor.core.publisher.MonoFlatMap$FlatMapMain.onComplete(MonoFlatMap.java:189)
        at reactor.core.publisher.MonoCreate$DefaultMonoSink.success(MonoCreate.java:140)
        at io.asyncer.r2dbc.mysql.client.ReactorNettyClient.lambda$close$11(ReactorNettyClient.java:194)
        at reactor.core.publisher.MonoCreate.subscribe(MonoCreate.java:58)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4485)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:263)
        at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51)
        at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64)
        at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4485)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:263)
        at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4485)
        at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onError(MonoIgnoreThen.java:278)
        at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106)
        at reactor.core.publisher.Operators.error(Operators.java:198)
        at reactor.core.publisher.MonoError.subscribe(MonoError.java:53)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4485)
        at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103)
        at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124)
        at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124)
        at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.onError(FluxTimeout.java:219)
        at reactor.core.publisher.MonoIgnoreElements$IgnoreElementsSubscriber.onError(MonoIgnoreElements.java:84)
        at reactor.core.publisher.FluxHandle$HandleSubscriber.onError(FluxHandle.java:212)
        at reactor.core.publisher.FluxHandle$HandleSubscriber.onNext(FluxHandle.java:134)
        at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2545)
        at reactor.core.publisher.FluxHandle$HandleSubscriber.request(FluxHandle.java:271)
        at reactor.core.publisher.MonoIgnoreElements$IgnoreElementsSubscriber.onSubscribe(MonoIgnoreElements.java:72)
        at reactor.core.publisher.FluxHandle$HandleSubscriber.onSubscribe(FluxHandle.java:89)
        at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55)
        at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4485)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:263)
        at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51)
        at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64)
        at io.r2dbc.pool.MonoDiscardOnCancel.subscribe(MonoDiscardOnCancel.java:50)
        at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:165)
        at reactor.pool.AbstractPool$Borrower.deliver(AbstractPool.java:467)
        at reactor.pool.SimpleDequePool.lambda$drainLoop$8(SimpleDequePool.java:368)
        at reactor.core.scheduler.ImmediateScheduler.schedule(ImmediateScheduler.java:52)
        at reactor.pool.SimpleDequePool.drainLoop(SimpleDequePool.java:368)
        at reactor.pool.SimpleDequePool.pendingOffer(SimpleDequePool.java:598)
        at reactor.pool.SimpleDequePool.doAcquire(SimpleDequePool.java:294)
        at reactor.pool.AbstractPool$Borrower.request(AbstractPool.java:430)
        at reactor.core.publisher.MonoFlatMap$FlatMapMain.request(MonoFlatMap.java:194)
        at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2341)
        at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2215)
        at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:117)
        at reactor.pool.SimpleDequePool$QueueBorrowerMono.subscribe(SimpleDequePool.java:716)
        at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64)
        at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52)
        at reactor.core.publisher.FluxRetry$RetrySubscriber.resubscribe(FluxRetry.java:117)
        at reactor.core.publisher.FluxRetry$RetrySubscriber.onError(FluxRetry.java:101)
        at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:241)
        at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:315)
        at io.r2dbc.pool.MonoDiscardOnCancel$MonoDiscardOnCancelSubscriber.onError(MonoDiscardOnCancel.java:98)
        at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onError(MonoIgnoreThen.java:278)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:231)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:203)
        at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete(FluxDoFinally.java:128)
        at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete(FluxDoFinally.java:128)
        at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.onComplete(MonoIgnoreThen.java:209)
        at reactor.netty.FutureMono$FutureSubscription.operationComplete(FutureMono.java:196)
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
        at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
        at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
        at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:625)
        at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:105)
        at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:990)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:686)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:620)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.close(DefaultChannelPipeline.java:1352)
        at io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:749)
        at io.netty.channel.AbstractChannelHandlerContext.access$1200(AbstractChannelHandlerContext.java:61)
        at io.netty.channel.AbstractChannelHandlerContext$11.run(AbstractChannelHandlerContext.java:732)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:403)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1623)
Caused by: io.r2dbc.spi.R2dbcNonTransientResourceException: Connection validation failed
        at io.r2dbc.pool.Validation.lambda$validate$2(Validation.java:45)
        at reactor.core.publisher.FluxHandle$HandleSubscriber.onNext(FluxHandle.java:112)
        ... 61 common frames omitted
```
Can you please help with this as currently I have to put retry on every DB call so that production does not break? Please let me know in case anything is not clear or I am missing something on my end.

**Comment by JohnNiang** [comment]
Hi @ssharan27 , which version do you use? If you haven't used [1.0.2](https://github.com/asyncer-io/r2dbc-mysql/releases/tag/r2dbc-mysql-1.0.2), please try it.

**Comment by ssharan27** [comment]
Hi @JohnNiang , currently I am using `1.0.0`, will check with `1.0.2` and see if it appears again. Thanks!

**Comment by ssharan27** [comment]
Hi @JohnNiang , @jchrys  still facing this issue. I checked the logs carefully and I was able to reproduce this on local as well. Should I create a new issue for this?

**Comment by jchrys** [comment]
Please create it as a new issue.
it would be greater if you include a reproducer.
Thank you for your input and cooperation 👍

**Comment by ssharan27** [comment]
Hi @jchrys, created a [new issue](https://github.com/asyncer-io/r2dbc-mysql/issues/128), please take a look.
